### ISSUE TITLE
[Animated] Interpolation between hex-encoded colors is unsupported

### ISSUE THREAD

**Issue by ide** [facebook/react-native#1905]
Ex: setting the output range to `['#fff', '#2966fb']`.

**Comment by brentvatne** [comment]
Yeah it looks like just rgba is supported right now, this would be nice to have though

**Comment by brentvatne** [comment]
[d3-interpolate](https://github.com/d3/d3-interpolate) combined with [d3-color](https://github.com/d3/d3-color) accomplishes this nicely, perhaps we should just pull d3-color in as a dependency?

**Comment by brentvatne** [comment]
[License is a bit weird though...](https://github.com/d3/d3-color/blob/master/LICENSE#L14-L16)

**Comment by nicklockwood** [comment]
cc: @vjeux

**Comment by vjeux** [comment]
The way string interpolation is implemented right now is by looking for consecutive numbers and interpolating those. Super dumb but works okay. Would be nice to have a proper color parser :) Wanna send a pull request?

**Comment by brentvatne** [comment]
@vjeux - are you OK with depending on d3-color? Should be easy if so

**Comment by vjeux** [comment]
i'd rather not. We already have color extraction in obj-c, java and this would add another way to interpret colors in js.

What would be nice is to move color normalization in js this way we can share the logic.

In the meantime, can you just add special support for #xxx and #xxxxxx?

**Comment by nicklockwood** [comment]
Maybe we could add a Color proptype instead of using strings for colors in js? then we could move all string->color conversion to the front-end, and send colors over the bridge as an array of numbers.

**Comment by nicklockwood** [comment]
To be clear, I'm not suggesting we'd stop using hex strings in the JS apps - just pre-convert them to an array before sending over bridge.

**Comment by aleclarson** [comment]
:+1: How important is this to you guys?

**Comment by ide** [comment]
Support will be added probably when someone contributes a PR.

**Comment by aleclarson** [comment]
Do you guys not use React Native for animations? Or just using RGBA values?

**Comment by ide** [comment]
I think all the Facebook examples have used RGBA values.

**Comment by ide** [comment]
Color processing is now in JS so this should be easier to add (or maybe just works? haven't checked).

**Comment by dralletje** [comment]
This one is fixed now? ^^ #3177
